docs: Shorten blog tutorial (#3763)
ignite/cli: docs/docs/02-guide/04-blog.md

---
description: Explore the essentials of module development while creating a dynamic blogging platform on your blockchain, where users can seamlessly submit and access blog posts, gaining practical experience in decentralized application functionalities.
title: Blog tutorial
---

# Build a Blog on a Blockchain with Ignite CLI

## Introduction

This tutorial guides you through creating a blog application as a Cosmos SDK blockchain using Ignite CLI. You'll learn to set up types, messages, queries, and write logic for creating, reading, updating, and deleting blog posts.

## Creating the Blog Blockchain

1. **Initialize the Blockchain:**

> ```bash

Reviewer: ```suggestion

```bash
```

ignite scaffold chain blog
cd blog
```

2. **Define the Post Type:**

> ```bash

Reviewer: ```suggestion

```bash
```

ignite scaffold type post title body creator id:uint
```

> This step creates a Post type with title, body (both strings), creator (string), and id (unsigned integer).

Reviewer: ```suggestion

This step creates a Post type with title, body (both strings), creator (string), and id (unsigned integer).
```

> This step creates a Post type with title, body (both strings), creator (string), and id (unsigned integer).

Reviewer 2: ```suggestion
This step creates a Post type with title (string), body (string), creator (string), and id (unsigned integer) fields.
```

## Implementing CRUD operations

**Creating Posts**

1. **Scaffold Create Message**

> ```bash

Reviewer: ```suggestion

```bash
```

ignite scaffold message create-post title body --response id:uint
```

> This message allows users to create posts with a title and body.

Reviewer: ```suggestion

This message allows users to create posts with a title and body.
```
